libcxx/include/__config

```diff
@@ -1215,6 +1215,21 @@ typedef __char32_t char32_t;
 #    define _LIBCPP_HAS_EXPLICIT_THIS_PARAMETER 0
 #  endif
 
+// TODO(LLVM 22): Remove this macro once LLVM19 support ends.
+#  if __has_builtin(__reference_constructs_from_temporary) ||                                                          \
+      (defined(_LIBCPP_CLANG_VER) && (_LIBCPP_CLANG_VER >= 1900))
```

Comment:
No change requested. I still think that this should be `_LIBCPP_CLANG_VER >= 1800`. But as support for Clang 18 will be dropped in LLVM 21 release, so the difference only affects CI, IIUC.

Reply:
`#if __has_builtin(__reference_constructs_from_temporary)" works correctly in Clang 18, so I think the above expresses the workaround more clearly.
https://godbolt.org/z/f8a1MG1Gq
If you think it's better to do it as you suggest. I'll do it. No problem.

I think the only case that's not handled completely currently is `__reference_constructs_from_temporary` on Android since we don't have a compiler that implements it on that platform and I suspect that the next NDK won't include the support too.

Reply:
Yeah. The current conditions look fine to me with your explanation.